traitsui/testing/tester/wx/tests/test_helpers.py

```diff
@@ -120,11 +120,40 @@ def test_key_sequence_disabled(self):
 
     def test_key_click(self):
         textbox = wx.TextCtrl(self.frame)
+        handler = mock.Mock()
+        textbox.Bind(wx.EVT_TEXT, handler)
 
         helpers.key_click_text_ctrl(textbox, command.KeyClick("A"), 0)
+
         self.assertEqual(textbox.Value, "A")
+        self.assertEqual(handler.call_count, 1)
+
+    def test_key_click_backspace(self):
+        textbox = wx.TextCtrl(self.frame)
+        textbox.SetValue("A")
+        handler = mock.Mock()
+        textbox.Bind(wx.EVT_TEXT, handler)
+
         helpers.key_click_text_ctrl(textbox, command.KeyClick("Backspace"), 0)
+
         self.assertEqual(textbox.Value, "")
+        self.assertEqual(handler.call_count, 1)
+
+    def test_key_click_backspace_with_selection(self):
+        textbox = wx.TextCtrl(self.frame)
+        textbox.SetFocus()
+        textbox.SetValue("ABCDE")
+        textbox.SetSelection(0, 4)
+        # sanity check
+        self.assertEqual(textbox.GetStringSelection(), "ABCD")
```

Comment:
Pedantic: This is not really a test objective so if the setup is not what we expect, we should raise a different exception to cause the test to "error", not to "fail".  I am not a purist and the comment explains the intent of this assertion already so I am going to leave it here. Happy to remove it too if others find this confusing.